scripts/variantstore/wdl/GvsExtractCallset.wdl

```diff
@@ -362,3 +374,56 @@ task CreateManifest {
     cpu: 1
   }
 }
+
+task GenerateSampleListFile {
+  # should not call cache in case of withdrawn samples
+  meta {
+    volatile: true
+  }
+
+  input {
+    String fq_samples_to_extract_table
+    String query_project
+
+    String? output_gcs_dir
+    String? service_account_json_path
+  }
+
+  String has_service_account_file = if (defined(service_account_json_path)) then 'true' else 'false'
+
+  command <<<
+    set -e
+
+    if [ ~{has_service_account_file} = 'true' ]; then
+      gsutil cp ~{service_account_json_path} local.service_account.json
+      gcloud auth activate-service-account --key-file=local.service_account.json
+      gcloud config set project ~{query_project}
+    fi
+
+    # Drop trailing slash if one exists
+    OUTPUT_GCS_DIR=$(echo ~{output_gcs_dir} | sed 's/\/$//')
+
+    echo "project_id = ~{query_project}" > ~/.bigqueryrc
+
+    bq --location=US --project_id=~{query_project} --format=csv query --use_legacy_sql=false "SELECT sample_name FROM ~{fq_samples_to_extract_table}" | sed 1d > sample-name-list.txt
+
+    if [ -n "$OUTPUT_GCS_DIR" ]; then
+      if [ ~{has_service_account_file} = 'true' ]; then
+        gsutil cp ~{service_account_json_path} local.service_account.json
+        gcloud auth activate-service-account --key-file=local.service_account.json
+      fi
```

Comment:
I'm confused about why this is necessary to do twice?

Reply:
it isn't — good catch!